packages/samples/headless-commerce-react/src/components/search-box/search-box.tsx

```diff
@@ -15,31 +21,65 @@ export default function SearchBox(props: ISearchBoxProps) {
     controller.subscribe(() => setState(controller.state));
   }, [controller]);
 
+  const onSearchBoxInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    controller.updateText(e.target.value);
+    instantProductsController.updateQuery(e.target.value);
+  };
+
   return (
     <div className="Searchbox">
       <input
         className="SearchBoxInput"
         value={state.value}
-        onChange={(e) => controller.updateText(e.target.value)}
+        onChange={(e) => onSearchBoxInputChange(e)}
       ></input>
       {state.value !== '' && (
         <span className="SearchBoxClear">
           <button onClick={controller.clear}>X</button>
         </span>
       )}
       <button onClick={controller.submit}>Search</button>
-      {state.suggestions.length > 0 && (
-        <ul className="QuerySuggestions">
-          {state.suggestions.map((suggestion, index) => (
-            <li key={index} className="QuerySuggestion">
-              <button
-                onClick={() => controller.selectSuggestion(suggestion.rawValue)}
-                dangerouslySetInnerHTML={{__html: suggestion.highlightedValue}}
-              ></button>
-            </li>
-          ))}
-        </ul>
-      )}
+      <table>
+        <thead>
+          <tr>
+            <th>Query suggestions</th>
+            <th>Instant products</th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr>
+            <td>
+              {state.suggestions.length > 0 && (
+                <ul className="QuerySuggestions">
+                  {state.suggestions.map((suggestion, index) => (
+                    <li key={index} className="QuerySuggestion">
```

Comment:
Should we use the `rawValue` as key actually? It should be unique. An `index` could be the same for two different suggestions, on different queries submitted to the search box.
```suggestion
                    <li key={rawValue} className="QuerySuggestion">
```

Reply:
Cannot be the same @Spuffynism , see [📖](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map#callbackfn) `Array.prototype.map`

Reply:
What I mean is that two different arrays share indices for items across renders. Using the `rawValue` would guarantee unique items, for unique values.

```
// first render, for query "cat"
// 0       1              2
  ["cats", "caterpillar", "category"]

// second render, for query "cate"
// 0              1
  ["caterpillar", "category"]
```

> A well-chosen key provides more information than the position within the array. Even if the position changes due to reordering, the key lets React identify the item throughout its lifetime.

https://react.dev/learn/rendering-lists#why-does-react-need-keys

Reply:
Yep, I'll use better keys